app/models/target.rb

```diff
@@ -48,4 +54,19 @@ def validate_target_limit
 
     errors.add(:targets, I18n.t('model.targets.errors.to_many'))
   end
+
+  def notify_compatible
+    users = near_targets.map(&:user)
+
+    NotificationService.new.send_compatible_target(users, self) unless users.empty?
+  end
+
+  def near_targets
+    near_targets = []
+    Target.where(topic: topic).where.not(user_id: user_id).each do |target|
```

Comment:
Checkout [geospacial database queries](https://github.com/alexreisner/geocoder#geospatial-database-queries) this will let you query targets at a database level. It will perform much faster than in memory.